compat/src/index.js

```diff
@@ -406,6 +404,16 @@ options.vnode = vnode => {
 		vnode.props.ref = vnode.ref;
 		vnode.ref = null;
 	}
+
+	// We can't just patch the base component class, because components that use
+	// inheritance and are transpiled down to ES5 will overwrite our patched
+	// getters and setters. See #1941
+	if (typeof type === 'function' && !type._patchedLifecycles && type.prototype) {
+		setSafeDescriptor(type, 'componentWillMount');
```

Comment:
A thought: passing `type.prototype` here could save a couple bytes and property lookups.

Reply:
Tried it out, but it looks like gzip already plays in our favor. Using a variable to store the `prototype` will lead to the same size and passing it directly is indeed `1B` smaller :tada: